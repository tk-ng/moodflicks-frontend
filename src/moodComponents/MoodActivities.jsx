import { Row, Col, ListGroup, ListGroupItem } from "reactstrap";
import { Link } from "react-router";

export const MoodActivities = ({ activities }) => {
	return (
		<Row>
			<Col xs="0" md="2" lg="3"></Col>
			<Col>
				<ListGroup>
					{activities.map((m) => {
						const key = m.moodID + m.movieID;
						return (
							<ListGroupItem
								key={key}
								color="danger"
								tag={Link}
								to={`/moods/${m.moodID}`}
							>
								<div className="text-dark">
									{m.createdBy} added a movie to their mood list "{m.mood}"{" "}
									{m.createdAgo}.
								</div>
							</ListGroupItem>
						);
					})}
				</ListGroup>
			</Col>
			<Col xs="0" md="2" lg="3"></Col>
		</Row>
	);
};
