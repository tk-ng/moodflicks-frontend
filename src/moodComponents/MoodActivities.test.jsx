import { render, screen } from "@testing-library/react";
import { MoodActivities } from "./MoodActivities";
import { MemoryRouter } from "react-router";
import { describe, expect } from "vitest";

const testActivities = [
	{
		movieID: 1111111,
		moodID: "01",
		mood: "Test Mood 1",
		createdBy: "testuser1",
		createdAgo: "3 days ago",
	},
	{
		movieID: 2222222,
		moodID: "02",
		mood: "Test Mood 2",
		createdBy: "testuser2",
		createdAgo: "5 days ago",
	},
	{
		movieID: 3333333,
		moodID: "03",
		mood: "Test Mood 3",
		createdBy: "testuser3",
		createdAgo: "7 days ago",
	},
];

describe("Render the MoodActivities component", function () {
	it("mounts without crashing", async function () {
		const { container, asFragment } = render(
			<MemoryRouter>
				<MoodActivities activities={[]} />
			</MemoryRouter>
		);
		expect(container).toBeInTheDocument();
		expect(asFragment).toMatchSnapshot();
	});

	it("render the component with activities", async function () {
		render(
			<MemoryRouter>
				<MoodActivities activities={testActivities} />
			</MemoryRouter>
		);
		const screenText1 = `testuser1 added a movie to their mood list "Test Mood 1" 3 days ago`;
		expect(screen.getByText(screenText1, { exact: false })).toBeInTheDocument();
	});
});
