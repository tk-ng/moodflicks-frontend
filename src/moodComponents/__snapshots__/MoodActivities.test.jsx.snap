// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render the MoodActivities component > mounts without crashing 1`] = `[Function]`;
